nix/mode.nix

```diff
@@ -0,0 +1,21 @@
+{
+  pkgs,
```

Comment:
I think we should just directly pass `lib` here, since we don't actually require anything else from `pkgs` in this file. Wherever this is used, you can just pass `pkgs.lib` if there is no other lib available.